presentation/main/src/main/java/com/hilingual/presentation/main/MainScreen.kt

```diff
@@ -67,17 +67,22 @@ private const val EXIT_MILLIS = 3000L
 
 @Composable
 internal fun MainScreen(
-    navigator: MainNavigator = rememberMainNavigator(),
-    viewModel: MainViewModel = hiltViewModel()
+    networkMonitor: NetworkMonitor,
+    appState: MainAppState = rememberMainAppState(networkMonitor = networkMonitor)
```

Comment:
P1: networkMonitor를 rememberMainAppState(networkMonitor = networkMonitor) 여기에서만 사용하는 것 같은데 그렇다면 state를 activity에서 받아오는 것이 아닌 MainScreen 내부에서 기본값으로 설정하신 이유가 있으신가요? 내부에서 사용하는 값이 아닌데 스크린이 받아오는 것이 조금 마음에 걸려서요..!!

Reply:
```suggestion
    appState: MainAppState
```

이렇게만 받아오고 MainActivity에서 state를 넣어주는 구조도 고려해보셨을까요ㅕ?? 테스트 코드를 추가할지는 모르겠지만 이런 구조가 조금 더 테스트 하기 용이할 것 같아서요! networkMonitor를 Screen 내부에서 사용하는 것이 아니라면 이런 방식을 고려해보는 것도 좋을 것 같아서 질문 드립니다!

Reply:
좋은 의견 감사합니다ㅎㅎ 그러게요...이렇게 안바꿀 이유가 없네요ㅎㅎ 상태 호이스팅해놓겠습니다ㅎㅎ